platform/lang-impl/src/com/intellij/execution/runners/AbstractConsoleRunnerWithHistory.java

```diff
@@ -147,29 +141,7 @@ protected Icon getConsoleIcon() {
   }
 
   protected String constructConsoleTitle(final @NotNull String consoleTitle) {
-    if (shouldAddNumberToTitle()) {
```

Comment:
1) Yes,  absolutely. I was planning to rebase and add issue numbers to the commit messages when done
2) I did moved 'constructConsoleTitle' to a seperate class to get rid of the inheritance from 'AbstractConsoleRunnerWithHistory' . It's actually a set of 3 methods "constructConsoleTitle", "shouldAddNumberToTitle",  and "getActiveConsoles".